faker/providers/internet/__init__.py

```diff
@@ -165,8 +165,22 @@ def domain_word(self,):
     def tld(self):
         return self.random_element(self.tlds)
 
-    def url(self):
-        pattern = self.random_element(self.url_formats)
+    def _filter_url_formats(self, exclude_scheme):
+        if not isinstance(exclude_scheme, str):
+            raise ValueError('exclude_scheme param should be a string')
+        if exclude_scheme.lower() not in ['https', 'http']:
+            raise ValueError('exclude_scheme param should either `http` or `https`')
+
+        pattern = '{}://'.format(exclude_scheme)
+        return filter(lambda f: not f.startswith(pattern), self.url_formats)
+
+    def url(self, exclude_scheme=None):
```

Comment:
I would rather have this more flexible and change the signature to something like:

```python
def url(self, schemes=None):
    if schemes is None:
        schemes = ['http', 'https']
```
This way people can generate any kind of scheme.

Reply:
Sure, I'm totally fine with that :)
I thought `exclude_scheme` were reasonable since actually only `http` and `https` schemes are generated.

I'm going to work on that asap, but I've a couple of questions.
What the expected behavior for the following scenarios?
```python
fake = Faker()

# One or all scheme/s does not "exists". 
fake.url(schemes=['foo', 'http'])   # foo://my-bar.org or error?
fake.url(schemes=['foo', bar'])

# Duplicates, raise an error?
fake.url(schemes=['foo', foo'])
fake.url(schemes=['http', 'http'])
```

Reply:
Why does `Faker` need to care about which schemes are real or not?

Reply:
@ProvoK:

* `schemes=['foo', bar']` will generate urls with `foo://` or `bar://`. As @swizzard say, `Faker` does't need to care if the scheme is real or not.
* `schemes=['foo', foo']` normalizes to a Set, so somehting like `schemes = Set(schemes)`

Reply:
@swizzard 
Actually, I was questioning about it since i was not sure what would be the best choice. If the generation of the url/email/whatever is too much permissive, a random string could be the same and there would be not the need of faker. However this seems not the case. :)

@fcurella 
got it!

thanks guys :+1:

Reply:
@fcurella 
I've done changes as requested. Tell me if there is something more i should/can do :smile: